wordpress-mobile/WordPress-Android
# Dashboard Personalization: Implements the Personalization Screen

## Issue
Part of #18940 and Closes #18942

## Description 
This PR implements the Personalization screen for dashboard cards


|  Light | Dark   | 
|---|---|
| ![Personalization_Screen_light](https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/d8709ef0-99aa-4cf1-8095-72b7be91f186)  | ![personalization_screen_dark](https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/83b2186d-7664-4932-8a56-a792b78e8dae)   | 

Note: the tracking will be taken up in a seperate PR 

## To test:
1. Go to Jetpack app 
2. Go to Me → Debug Settings → Enable `dashboard_personalization`
3. Go to the bottom of dashboard
4. Click on `Personalize your home tab`
5. ✅  Verify that the visibility of the cards are shown correct 
6. Toggle the visibility of any card 
7. Go to dashboard 
8. Verify that the card is hidden on Dashboard 
9. Repeat steps 6-8 for each card 


## Regression Notes
1. Potential unintended areas of impact
Dashboard cards are not hidden/shown properly

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and Unit testing

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:

UI Changes testing checklist:
